R/graph.R

```diff
@@ -0,0 +1,73 @@
+orderly_graph_packets <- function(from = NULL, to = NULL,
+                                  root = NULL, locate = FALSE) {
+  root <- root_open(root, locate = locate, require_orderly = FALSE,
+                    call = environment())
+  if (is.null(from) == is.null(to)) {
+    cli::cli_abort("Exactly one of 'from' and 'to' must be given")
+  }
+
+  metadata <- root$index$data()$metadata
+  if (!is.null(from)) {
+    dat <- graph_packets_from(from, metadata)
+  } else {
+    dat <- graph_packets_to(to, metadata)
+  }
+
+  dat
+}
+
+
+graph_packets_to <- function(to, metadata) {
+  validate_outpack_id(to)
```

Comment:
Should we test that this `id` actually exists? Get a pretty ugly error if it metadata[[p]]$depends is `NULL`

```
Error in if (nrow(depends) > 0) { : argument is of length zero
```

Reply:
yeah, these are both gross. I've added fixes and tests for both, thanks